### PR TITLE
[pt2] add meta for `linalg_ldl_factor_ex`

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #101375
* #101372
* #101367
* __->__ #101362
* #101315

### CODE REVIEW

**Comment by nkaretnikov** [comment]
Part of this stack because it unlocks tests next to the prev PR in the stack.

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -543,6 +543,30 @@ def linalg_inv_ex_meta(A: Tensor, check_errors: bool = False):
...
+    LD = self.new_empty(shape)
+    LD.as_strided_(shape, make_contiguous_strides_for(shape, row_major=False))
+
+    pivots = self.new_empty(shape[:-1], dtype=torch.int)
+
+    info = self.new_empty(shape[:-2], dtype=torch.int)
+
+    return LD, pivots, info
+
+
```
Comment (nkaretnikov) [inline torch/_meta_registrations.py:569]: See `TORCH_META_FUNC(linalg_ldl_factor_ex)` in core.

File: torch/_meta_registrations.py [hunk 2]
```diff
@@ -543,6 +543,30 @@ def linalg_inv_ex_meta(A: Tensor, check_errors: bool = False):
...
+    check_errors: bool = False,
+) -> Tuple[Tensor, Tensor, Tensor]:
+    squareCheckInputs(self, "torch.linalg.ldl_factor_ex")
+    checkFloatingOrComplex(self, "torch.linalg.ldl_factor_ex")
+
+    shape = self.shape
+
+    # prefer column major strides
+    LD = self.new_empty(shape)
+    LD.as_strided_(shape, make_contiguous_strides_for(shape, row_major=False))
```
Comment (lezcano) [inline torch/_meta_registrations.py:561]: It's not that we prefer them, but LAPACK and cusolver ask for the inputs to be column_major. Also, same as in the other PR, why not `empty_strided`?

**Review by lezcano — COMMENTED** [review]
Just a minor, otherwise this looks good

**Review by lezcano — APPROVED** [review]

**Pull request closed without merging**